plugins/techdocs/src/reader/transformers/liftMetaRefresh.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Transformer } from './transformer';
+import { normalizeUrl } from './rewriteDocLinks';
+
+export const liftMetaRedirects = (): Transformer => {
+  return dom => {
+    for (const elem of Array.from(dom.querySelectorAll('meta'))) {
+      if (elem.getAttribute('http-equiv') === 'refresh') {
+        const metaContentParameters = elem
+          .getAttribute('content')
+          ?.split('url=');
+        if (!metaContentParameters || metaContentParameters.length < 2) {
+          continue;
+        }
+        const relUrl = metaContentParameters[1];
+        const normalizedCurrentUrl = normalizeUrl(window.location.href);
+        const absoluteRedirect = new URL(relUrl, normalizedCurrentUrl).href;
+
+        const newContentValue = metaContentParameters[0].includes(';')
+          ? `${metaContentParameters[0]}url=${absoluteRedirect}`
+          : `url=${absoluteRedirect}`;
+        elem.setAttribute('content', newContentValue);
+
+        // Lift the meta tag from shadowDom to the main document
+        document.body.appendChild(elem);
```

Comment:
Interesting! I wouldn't have thought that dynamically appending meta tags representing a redirect would work in a SPA like this.

Is there a benefit to doing it this way? Would the browser actually cache the redirect?

My instinct would be to just use a `react-router`'s `navigate` function (passed in as an arg to this code here). The benefit this way would be that a redirect wouldn't incur a full page reload (re-downloading and parsing all of Backstage's bundles), just a normal route change.

Reply:
Hi Eric, I don't think there is a benefit to justify the full page reload.  I misinterpreted your RFC comment to be about using the meta tags to refresh. Other than consistency with how the `mkdocs-plugin` performs redirects, there is no benefit over using `navigate` in an SPA. 

Does get me thinking though, we really only need the relative url from the redirect map. The benefit of using `mkdocs-redirects` is that it stores this value in the html page generated for the legacy document vs us needing to define some other storage solution and retrieval method for the redirect map. Internally, we decorate our TechDoc metadata files with the content of the `mkdocs.yaml`. With this information, we could manage the redirects without using the plugin at all. However, I discarded this idea initially since it seems to be a non-standard pattern. This information in the `mkdocs.yaml `would never be used at the document generation stage and only exist to determine final display of the site in Backstage .

Reply:
Updated to use `navigate`